### DISCUSSION TITLE
Editor fullscreen / size on start-up

### DISCUSSION THREAD

**Discussion by som45oul** [godotengine/godot-proposals#10722] [Editor]
I know this has been a thing in the past. Not sure why the editor was made fullscreen by default, with no option to change that.  Is there any chance the ability to modify the start-up size could be added back in future?

Personally, I would like to see:
* Use size from last session
* Fixed size / state (maximised, minimised, etc)

Ideally, this would be an "Editor Settings" setting, but might also be good to have it as a command-line argument.  I think it was a command line argument in the past?

I get that some people want it fullscreen by default.  But I have a large widescreen and would prefer Godot to take up only half my screen when I start it up.  Having fullscreen as the default can make sense - but it's always better to have options in my view.

Thanks.

**Comment by tetrapod00** [comment]
I think this might have just been implemented in https://github.com/godotengine/godot/pull/76085? You can try it out in 4.4dev2.

**Reply by som45oul** [reply]
That's great to hear. Thanks for the heads up.

While I think saving and restoring the state is a good step forward, I would also like to see the ability to set the window size from the command-line in future.  Perhaps the author can build on that?  I will add some comments.

**Reply by Calinou** [reply]
> I would also like to see the ability to set the window size from the command-line in future.

This is already feasible: [Command line tutorial](https://docs.godotengine.org/en/stable/tutorials/editor/command_line_tutorial.html)

If these don't work on the editor or project manager, it should be reported as a bug.
